timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -243,22 +251,35 @@ def fit(
         model: AbstractTimeSeriesModel
             The fitted model object
         """
+        start_time = time.monotonic()
         self.initialize(**kwargs)
         self.target_scaler = self._create_target_scaler()
         if self.target_scaler is not None:
             train_data = self.target_scaler.fit_transform(train_data)
 
+        self.covariates_regressor = self._create_covariates_regressor()
+        if self.covariates_regressor is not None:
+            train_data = self.covariates_regressor.fit_transform(
+                train_data,
+                time_limit=time_limit * 0.5 if time_limit is not None else time_limit,
+            )
+
         train_data = self.preprocess(train_data, is_train=True)
         if self._get_tags()["can_use_val_data"] and val_data is not None:
             if self.target_scaler is not None:
                 val_data = self.target_scaler.transform(val_data)
+            if self.covariates_regressor is not None:
+                val_data = self.covariates_regressor.transform(val_data)
```

Comment:
You are right, but I'm not quite sure how to set the time_limit for `fit_transform` in this case. In practice, I don't think it will ever be relevant since only GluonTS models even use `val_data` during `fit()`, and for those model we probably don't want to use the "regressor" anyway.

The real "refit on val_data" will happen during `predict` for all models when they call `score_and_cache_oof`.